### PR TITLE
Minimize builds & Supervisor in Docker

### PR DESCRIPTION
**Pull request by Bellardia**
This PR reformulates vbox and docker containers to create much smaller base images. This is mostly done through better cleanup, removing unneeded dependencies and binaries, and deferring installation of some packages. Due to datadogs sheer size (almost 1gb for all dependencies), it's been removed from the base instance.

Secondly, containers move to using `supervisor` as a process manager, and the `service` command is extended to seamlessly operate on both `systemd` and `supervisor` systems. This allows us to continue to use the existing cookbook control flow while removing systemd from docker images.

Since systemd is removed, we no longer need privileged containers or cgroups.

### CODE REVIEW

File: base/recipes/gearman.rb [hunk 1]
```diff
@@ -73,10 +73,10 @@
   owner "root"
   group "root"
   mode "0755"
-  notifies :run, "execute[gearman systemcl daemon-reload]", :immediately
+  notifies :run, "execute[gearman systemctl daemon-reload]", :immediately unless node[:container]
```
Comment (ahosking) [inline base/recipes/gearman.rb:76]: @Bellardia I never had success with this though, it looks like my syntax may have been off, ever-so-slightly.

File: blog/recipes/default.rb [hunk 2]
```diff
@@ -69,10 +69,10 @@
     group: node[:user],
   })
   mode "0755"
-  notifies :run, "execute[blog systemcl daemon-reload]", :immediately
+  notifies :run, "execute[blog systemctl daemon-reload]" unless node[:container]
 end
 
-execute "blog systemcl daemon-reload" do
+execute "blog systemctl daemon-reload" do
```
Comment (ahosking) [inline blog/recipes/default.rb:75]: Good catch on the typos here.

File: configure/files/container.rb [hunk 3]
```diff
@@ -1,8 +1,9 @@
 Ohai.plugin(:Container) do
+  depends "etc/passwd"
   provides "container"
 
   collect_data(:default) do
-    if File.exist?("/.dockerenv") || !!(File.read("/proc/1/cgroup") =~ %r[^\d+:\w+:/(lxc|docker)/])
+    if File.exist?("/.dockerenv") || !!(File.read("/proc/1/cgroup") =~ %r[^\d+:\w+:/(lxc|docker)/]) || etc["passwd"].key?("gitpod")
```
Comment (ahosking) [inline configure/files/container.rb:6]: Do we need to keep things here specifically for `gitpod` as a service?

File: jbx/recipes/kinesis.rb [hunk 4]
```diff
@@ -19,6 +19,7 @@
           },
         }
       )
+      triggers_reload !node[:container]
```
Comment (ahosking) [inline jbx/recipes/kinesis.rb:22]: Shouldn't this become 
```suggestion
      triggers_reload  unless node[:container]
```

File: dnsmasq/recipes/default.rb [hunk 5]
```diff
@@ -2,7 +2,7 @@
 user "dnsmasq"
 
 if platform?("ubuntu") && node["lsb"]["release"].to_i >= 18
-  if !node.attribute?(:container)
+  unless node.attribute?(:container)
```
Comment (ahosking) [inline dnsmasq/recipes/default.rb]:
```suggestion
  unless node[:container]
```

File: configure/files/container.rb (hunk 3 above)
Comment (Bellardia) [reply configure/files/container.rb:6]: Yeah gitpod evades the other 2 docker checks

File: jbx/recipes/kinesis.rb (hunk 4 above)
Comment (Bellardia) [reply jbx/recipes/kinesis.rb:22]: Not in this case, `unless` is a conditional. We need to provide a boolean value to `triggers_reload`

**Pull request merged**